File: Deepspace-Ruby/lib/test_p1.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative 'loot'
require_relative 'supplies_package'
require_relative 'shield_booster'
require_relative 'weapon'
require_relative 'weapon_type'
require_relative 'dice'
require_relative 'game_character'


class TestP1
  def initialize
    
  end
  
  def main
    l=Loot.new(1, 2, 3, 4, 5)
    
    print "nSupplies", l.nSupplies, "nWeapons", l.nWeapons, "nShields", l.nShields, "nHangars", l.nHangars, "nMedals", l.nMedals
    shields= ShieldBooster.new("escudo", 2.0, 1)
    print "Boost", shields.boost,"Uses", shields.uses, "UseIt", shields.useIt
    su=SuppliesPackage.new(1,2,3)
    print "ammo", su.ammoPower,"fuel", su.fuelUnits, "shield", su.shieldPower
    misil=WeaponType::MISSILE
    weapon=Weapon.new("pistola", misil, 8)
    print "type", weapon.type, "uses", weapon.uses, "power",weapon.power, "useIt", weapon.useIt
    
    dado=Dice.new
    nShields=0
    nHangars=0
    nWeapons=0
    nWeapons2=0
    
    for i in 0..100 do
      if dado.initWithNShields==0
        nShields+=1
      end
      if dado.initWithNHangars==0
        nHangars+=1
      end
      if dado.initWithNWeapons==1
        nWeapons+=1
      elsif dado.initWithNWeapons==2
        nWeapons2+=1
      end
    end
    
    print "El numero de veces que se ha dado nShields es ", nShields, " cuando tiene que estar alrededor de 25."
    print "El numero de veces que se ha dado nHangars es ", nHangars, " cuando tiene que estar alrededor de 25."
    print "El numero de veces que se ha dado nWeapons es ", nWeapons, " cuando tiene que estar alrededor de 33"
    print "El numero de veces que se ha dado nWeapons2 es ", nWeapons2, " cuando tiene que estar alrededor de 33"
    dado.to_s
  end
  
end

nuevo=TestP1.new
nuevo.main

File: Deepspace-Java/src/deepspace/TestP1.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

/**
 *
 * @author inaki
 */
public class TestP1 {
    public static void main(String[] args){        
     
        
        WeaponType wt=WeaponType.MISSILE;
        Weapon w=new Weapon("escopeta", wt, 3);
        System.out.println("Tipo: "+ w.getType());
        System.out.println("Usos: "+ w.getUses());
        System.out.println("Potencia: "+ w.power());
        System.out.println("useIt: "+ w.useIt());
        
        SuppliesPackage su=new SuppliesPackage(1.0f,2.0f,3.0f);
        
        System.out.println("AmmoPower: " + su.getAmmoPower());
        System.out.println("fuelUnits: " + su.getFuelUnits());
        System.out.println("ShieldsPower: " + su.getShieldsPower());
        
        ShieldBooster sh=new ShieldBooster("escudo", 2.0f, 1);
        
        System.out.println("Boost: " + sh.getBoost());
        System.out.println("Uses: " + sh.getUses());
        System.out.println("useIt" + sh.useIt());
        
        Loot l=new Loot(1, 2, 3, 4, 5, false, false);
        
        System.out.println("nSupplies: " + l.getNSupplies());
        System.out.println("nWeapons: " + l.getNWeapons());
        System.out.println("nShields: " + l.getNShields());
        System.out.println("nHangars: " + l.getNHangars());
        System.out.println("nMedals: " + l.getNMedals());
        
        Dice d=new Dice();
        
        System.out.println("FirstShot: " + d.firstShot());
        System.out.println("InitWithNShields: " + d.initWhithNShields());
        System.out.println("InitWithNHangars: " + d.initWithNHangars());
        System.out.println("initWithNWeapons: " + d.initWithNWeapons());
        System.out.println("spaceStation: " + d.spaceStationMoves(0.33f));
        System.out.println("whoStar: " + d.whoStarts(9));
        
        
        int NShields=0;
        int NHangars=0;
        int NWeapons=0;
        int NWeapons2=0;
        for (int i=0; i<100; i++){
            if (d.initWhithNShields()==0){
                NShields++;
            }
            
            if (d.initWithNHangars()==0){
                NHangars++;
            }
            
            if (d.initWithNWeapons()==1){
                NWeapons++;
            }
            
            if(d.initWithNWeapons()==2){
                NWeapons2++;
            }
        }
        
        System.out.println("El numero de veces que se da NSHIELDS es " + NShields + " cuando tendría que ser más o menos 25");
        System.out.println("El numero de veces que se da NHANGARSPROB es " + NHangars + " cuando tendría que ser más o menos 25");
        System.out.println("El numero de veces que se da NWEAPONSPROB es " + NWeapons + "cuando tendría que ser más o menos 33");
        System.out.println("El numero de veces que se da 2*NWEAPONSPROB es " + NWeapons2 + "cuando tendría que ser más o menos 33");
    }
    
       
}
